### ISSUE TITLE
LWFA on cpu shows different (wrong) physics

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/picongpu#2688]
When running the default LWFA example with `32.cfg`there is a major difference between using gpus or cpus.

In the following you will see png pictures of the simulation after 600 iterations/time steps. 

The current `master` brach:
![grafik](https://user-images.githubusercontent.com/5121158/44151879-25acdd80-a0a4-11e8-9b0e-da10f76ede44.png)

The current `dev`branch on `k80`GPUs:
![grafik](https://user-images.githubusercontent.com/5121158/44151888-32607ff0-a0a4-11e8-8ffd-5fb274d82953.png)

The current `dev`branch on `laser`CPUs:
![grafik](https://user-images.githubusercontent.com/5121158/44151904-3e8a0e04-a0a4-11e8-95eb-4cdef28dc2bd.png)

The last is definitely wrong, an `a_0=8.0` would cause a blow out, not these streams.

**Comment by PrometheusPi** [comment]
@kossag14 This is probably not an issue that effects you. I think it is more likely the difference you see arise from #2687.

**Comment by PrometheusPi** [comment]
as recommended by @psychocoderHPC *offline* I checked how PIConGPU performs on CPUs (`laser`nodes) at `60 commits` **before** `tags/0.4.0-rc1`. 

The results looks great! We have "correct" physics! ✨  🎉 

600 iterations of `57bcf5cc592b3bf9ad14902d1cadbd4ec8d76193` on `laser` nodes:
![grafik](https://user-images.githubusercontent.com/5121158/44155992-1653e850-a0af-11e8-9d14-08168c536731.png)


Probably the laser refactoring #2587 (or #2624) caused this bug.

**Comment by PrometheusPi** [comment]
@psychocoderHPC as you expected, calling the laser functor is causing the error.

With the fix:
```diff
diff --git a/include/picongpu/fields/LaserPhysics.hpp b/include/picongpu/fields/LaserPhysics.hpp
index 66f8c84..09bab6f 100644
--- a/include/picongpu/fields/LaserPhysics.hpp
+++ b/include/picongpu/fields/LaserPhysics.hpp
@@ -79,12 +79,6 @@ namespace fields
 
             uint32_t cellOffsetInSuperCellFromInitPlaneY = LaserFunctor::Unitless::initPlaneY % SuperCellSize::y::value;
 
-            auto accLaserFunctor = laserFunctor(
-                acc,
-                localSuperCellOffset,
-                mappings::threads::WorkerCfg< numWorkers >{ workerIdx }
-            );
-
             mappings::threads::ForEachIdx<
                 mappings::threads::IdxConfig<
                     planeSize,
@@ -96,6 +90,12 @@ namespace fields
                     uint32_t const
                 )
                 {
+                   auto accLaserFunctor = laserFunctor(
+                       acc,
+                       localSuperCellOffset,
+                       mappings::threads::WorkerCfg< numWorkers >{ workerIdx }
+                    );
+
                     /* cell index within the superCell */
                     DataSpace< simDim > cellIdxInSuperCell = DataSpaceOperations< simDim >::template map< LaserPlaneSizeInSuperCell >( linearIdx );
                     cellIdxInSuperCell.y() += cellOffsetInSuperCellFromInitPlaneY;
```
the issue is solved:

png picture after 600 iterations with above fix on `laser` cpu nodes:

![grafik](https://user-images.githubusercontent.com/5121158/44158825-05681780-a0b6-11e8-96d4-5a0e09e3e084.png)

**Comment by ax3l** [comment]
thx for the fix in #2689! will add it to `0.4.0-rc2`
